andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java

```diff
@@ -38,6 +41,21 @@ public boolean visit(MethodInvocation mi) {
         return super.visit(mi);
     }
 
+    @Override
+    public boolean visit(FieldDeclaration fd) {
+        boolean hasMockAnnotation = fd.modifiers().stream()
+                .anyMatch(m -> m instanceof Annotation &&
+                        ((Annotation)m).getTypeName().getFullyQualifiedName().equals("Mock"));
+
+        // If the field is annotated with @Mock, check if it's the class we are interested in
+        if (hasMockAnnotation) {
+            String className = fd.getType().toString();
+            classWasMocked = className.contains(classToBeMocked);
```

Comment:
If I understand the code correctly, `classWasMocked` could be erroneously set to false if there are multiple usages of `@Mock`.  For example, the following could happen:
1. A search for a mocked instance of `Set` is started
2. `@Mock Set<Integer> a;` is visited,  `classWasMocked` is set to true
3. `@Mock List<String> b;` is visited,  `classWasMocked` is set to false
4. The final returned value is false

I see that the other visitor prevents this problem by only executing the logic if `classWasMocked` is still false.

Reply:
Yeah, you're right. To handle that, I should add a `(!classWasMocked)` conditional. I will update the code and tests respectively.

Reply:
Updates committed and new test passes.

Reply:
Thanks! :)